packages/fxa-auth-server/lib/oauth/db/accessToken.js

```diff
@@ -87,31 +89,92 @@ class AccessToken {
       createdAt: this.createdAt.getTime(),
       profileChangedAt: this.profileChangedAt,
       expiresAt: this.expiresAt.getTime(),
-      token: this.token.toString('hex'),
     };
   }
 
+  static generate(
+    clientId,
+    name,
+    canGrant,
+    publicClient,
+    userId,
+    email,
+    scope,
+    profileChangedAt,
+    expiresAt,
+    ttl
+  ) {
+    const token = unique.token();
+    const tokenId = encrypt.hash(token);
+    return new AccessToken(
+      tokenId,
+      clientId,
+      name,
+      canGrant,
+      publicClient,
+      userId,
+      email,
+      scope,
+      // Truncated createdAt to the second to match mysql
+      new Date(new Date().setMilliseconds(0)),
+      profileChangedAt || 0,
+      expiresAt || new Date(Date.now() + (ttl * 1000 || MAX_TTL)),
+      // This is the one and only time the caller can get at the unhashed token.
+      token
+    );
+  }
+
   /**
    *
    * @param {string} string
    * @returns {AccessToken}
    */
   static parse(string) {
     const json = JSON.parse(string);
+    // We previously wrote the raw token into redis, but are now writing
+    // only the hashed tokenId.
+    let tokenId;
+    if (json.tokenId) {
+      tokenId = Buffer.from(json.tokenId, 'hex');
+    } else {
+      tokenId = encrypt.hash(Buffer.from(json.token, 'hex'));
+    }
     return new AccessToken(
+      tokenId,
       Buffer.from(json.clientId, 'hex'),
       json.name,
       json.canGrant,
       json.publicClient,
       Buffer.from(json.userId, 'hex'),
       json.email,
       ScopeSet.fromString(json.scope),
-      Buffer.from(json.token, 'hex'),
       new Date(json.createdAt),
       json.profileChangedAt,
       new Date(json.expiresAt)
     );
   }
+
+  /**
+   * @param {any} row
+   * @returns {AccessToken}
+   */
+  static fromMySQL(row) {
+    return new AccessToken(
+      // Somewhat confusingly, the `token` column in MySQL is the hashed value
+      // that we refer to as `tokenId`.
+      row.token,
```

Comment:
what if the query used `tokens.token AS tokenId` instead?

Reply:
👍 I'm going to this, and use it as an opportunity to document the *why* as well.